doc/user/inputs/blueprints.rst

```diff
@@ -404,9 +404,98 @@ material modifications
   you can define any fuel material as being made of LWR-derived TRU plus depleted uranium
   at various weight fractions. Note that this input style only adjusts the heavy metal.
 
-  .. warning:: The input processing system will try to apply the extra input parameters to all components in
-        the block, so there should typically only be one component per block that understands each input
-        parameter.
+  To enable the application of different values for the same material modification type
+  on different components within a block, the user may specify componentwise material
+  modifications. This is useful, for instance, when two pins within an assembly
+  made of the same base material have different fuel enrichments. This is done
+  using the `componentwise` attribute to the material modifications as in::
+
+        blocks:
+            fuel: &block_fuel
+                fuel1: &component_fuel_fuel1
+                    shape: Hexagon
+                    material: UZr
+                    Tinput: 600.0
+                    Thot: 600.0
+                    ip: 0.0
+                    mult: 1
+                    op: 10.0
+                fuel2: &component_fuel_fuel2
+                    shape: Hexagon
+                    material: UZr
+                    Tinput: 600.0
+                    Thot: 600.0
+                    ip: 0.0
+                    mult: 1
+                    op: 10.0
+        assemblies:
+            fuel a: &assembly_a
+                specifier: IC
+                blocks: [*block_fuel]
+                height: [1.0]
+                axial mesh points: [1]
+                xs types: [A]
+                material modifications:
+                    componentwise:
+                        fuel1:
+                            U235_wt_frac: [0.20]
+                        fuel2:
+                            Zr_wt_frac: [0.02]
+                    U235_wt_frac: [0.30]
+
+  This example would apply an enrichment of 20% to the `fuel1` component and an
+  enrichment of 30% to all other components in the block that accept the `U235_wt_frac`
+  material modification. The user may alternatively use the `blockwise` attribute
+  to more clearly indicate that the material modification type is being applied to
+  all applicable components in the block::
+
+                material modifications:
+                    componentwise:
```

Comment:
does `bycomponent`  and `byblock` sound better to you?  Or just `component` and `block`? The `wise` part in these names feels slightly funny to me. Not a big deal though.

Reply:
Yeah, I like `bycomponent` better. I'll probably change the key to `by component`, since yamlize is capable of that and it is more clear.

Still don't quite like `by block`, though. To me, it is not clear how the `by block` concept is different from how the material modifications are required to be entered as a list with one value per block. I'm thinking a name more like `block default` would work better. That strongly implies the concept that the material modification will fall back to the `block default` value if a `by component` value does not supersede it.

What do you think?

Reply:
I like the idea of just adding `by component` and then leaving the block material modifications at the top level. This will keep backwards compatibility with inputs.

Reply:
Yeah, that might be the best way to go. I always tend to like symmetry, so if there is a `by block` attribute, my gut wants there to be an alternative attribute as well. But just leaving the blockwise values on the `material modifications` level might be best.

Reply:
I changed the latest commit to use `by component` and got rid of the `blockwise` option entirely. Now all block default values are entered as they were before on the `material modifications` level.

Reply:
It looks to me like this conversation can be "resolve"d.  @keckler ?

Reply:
In my view it is resolved, just didn't mark it as such in case anyone else felt differently.